haystack/components/converters/csv.py

```diff
@@ -102,3 +105,231 @@ def run(
             documents.append(document)
 
         return {"documents": documents}
+
+
+@component
+class CSVDocumentSplitter:
+    """
+    A component to split a CSV document into multiple CSV documents
+
+    It is based on:
+    1) Consecutive empty rows (the "classic" approach).
+    2) Optionally detecting side-by-side tables (via a BFS search for non-empty cells).
+
+    Attributes:
+        split_threshold (int): The number of consecutive empty rows required to split the CSV. Default is 2.
+        delimiter (str): The delimiter used in the CSV. Default is ','.
+        quotechar (str): The quote character used in the CSV. Default is '"'.
+        trim_empty_rows (bool): Whether to trim leading/trailing empty rows in each block. Default is True.
+        skip_errors (bool): Whether to skip documents with parsing errors. Default is True.
+        detect_side_tables (bool): Whether to detect tables that sit side-by-side in different columns.
+                                   If True, a more expensive BFS-based approach is used.
```

Comment:
Let's move these to the docstrings of the init method to follow our doc patterns of other components. Also let's update it to follow 
```
:param [ParamName]: [ParamDescription]
```